.github/workflows/lint-and-test.yml

```
@@ -0,0 +1,40 @@
+name: Run Tests and Linter
+
+on:
+  pull_request:
+    branches:
+      - master
+      - main
+
+jobs:
+  lint-and-test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.8.18, 3.9.18, 3.10.13, 3.11.8]
+        django-version: [3.2, 4.2]
+
+    name: Lint and Test (Python ${{ matrix.python-version }} - Django ${{ matrix.django-version }})
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
```

```suggestion
          python-version: ${{ matrix.python-version }}
          cache: 'pip'
```

This will speed up subsequent runs, as long as you're cool with storing the dependencies in the GHA cache

---

I think that's a good call. However, it appears to look for a `requirements.txt` file for the dependencies which causes it to fail. I'm unsure how to get it to pickup the dependencies defined in `setup.py` since this is a python library repo

---

If you know how to do it, I'm definitely willing to add it. Otherwise, I am okay with the slow runs since this will rarely run